cpp/src/community/louvain.cu

```diff
@@ -28,6 +28,22 @@ namespace cugraph {
 
 namespace detail {
 
+template <typename vertex_t, typename edge_t, typename weight_t, bool multi_gpu>
+void check_clustering(
+  experimental::graph_view_t<vertex_t, edge_t, weight_t, false, multi_gpu> const& graph_view,
+  vertex_t* clustering)
+{
+  if (graph_view.get_number_of_local_vertices() > 0)
+    CUGRAPH_EXPECTS(clustering != nullptr, "Invalid input argument: clustering is null");
```

Comment:
Is this really an invalid use? I assume it is valid (even though hugely inefficient) to ask cuGraph to compute Louvain for a tiny graph using many GPUs (e.g. # GPUs > # vertices).

Reply:
If the number of local vertices is 0 then the `clustering` pointer is never referenced, nothing is required to be written.

The issue that triggered this is that if the graph is poorly distributed (karate over 8 GPUs apparently ends up with at least 1 cluster which has no vertices assigned to it) then the python code creates a cudf column with 0 entries.  This results in one of the calls using `clustering == nullptr`, which was failing.  This check will allow `clustering` to be `nullptr` if there are no vertices assigned to this GPU.  If there are vertices assigned to this GPU it will fail if `clustering` is `nullptr`

Reply:
Ah... I see. I misunderstood the code and thanks for the explanation.